hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -1257,7 +1258,8 @@ public String getSelectGUIDString() {
 	 * @return the SQL equivalent to Oracle's {@code from dual}.
 	 */
 	public String getFromDual() {
-		return "";
+		// The standard SQL solution to get a dual table is to use the VALUES clause
+		return "from (values (0)) as dual";
```

Comment:
IIRC, Postgres doesn't need this, so if you're going to add this here, please also override it on `PostgresDialect` to return `""`.

Right now I don't recall if there were other databases which also don't require that.

Reply:
If a query uses other clauses like `where`, `group by` etc. it needs a from clause. I can add another method `supportsSelectQueryWithoutFromClause` to handle these cases if you like.

Reply:
> If a query uses other clauses like `where`, `group by` etc. it needs a from clause. 

Ah, OK, right. But does it make sense to have these clauses if there is no `from` clause? I don't see how they could possibly be useful.

> I can add another method `supportsSelectQueryWithoutFromClause` to handle these cases if you like.

I would prefer that because `from (values (0)) as dual` is really pretty ugly and noisy.

Reply:
For the record, I needed this in the emulation for tuple comparison like `(a, b) in ((1, 2), (2, 3))` for DBMS that only support tuples for IN predicates with subqueries like e.g. Oracle. I'll add an additional method to handle omitting the from clause altogether though.

Reply:
OK, that makes sense. 👍